evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,365 @@
+            pipeline = pipeline_dict["pipeline"]
+            pipeline_params = pipeline_dict["parameters"]
+            parameters = self._transform_parameters(pipeline, pipeline_params)
+            input_pipelines.append(
+                pipeline.new(parameters=parameters, random_seed=self.random_seed)
+            )
+        n_jobs_ensemble = 1 if self.text_in_ensembling else self.n_jobs
+        ensemble = _make_stacked_ensemble_pipeline(
+            input_pipelines,
+            input_pipelines[0].problem_type,
+            random_seed=self.random_seed,
+            n_jobs=n_jobs_ensemble,
+        )
+        return [ensemble]
+
+    def _create_long_top_n(self, n):
+        estimators = [
+            (pipeline_dict["pipeline"].estimator, pipeline_dict["mean_cv_score"])
+            for pipeline_dict in self._best_pipeline_info.values()
+        ]
+        estimators.sort(key=lambda pipeline: pipeline[1])
+        estimators = estimators[:n]
+        estimators = [estimator[0].__class__ for estimator in estimators]
+        pipelines = [
+            make_pipeline(
+                self.X,
+                self.y,
+                estimator,
+                self.problem_type,
+                sampler_name=self._sampler_name,
+                extra_components=[SelectColumns],
+            )
+            for estimator in estimators
+        ]
+        self._top_n_pipelines = pipelines
+        next_batch = []
+        for _ in range(50):
+            for pipeline in pipelines:
+                if pipeline.name not in self._tuners:
+                    self._create_tuner(pipeline)
+                proposed_parameters = self._tuners[pipeline.name].propose()
+                parameters = self._transform_parameters(pipeline, proposed_parameters)
+                parameters.update(
+                    {"Select Columns Transformer": {"columns": self._selected_cols}}
+                )
+                next_batch.append(
+                    pipeline.new(parameters=parameters, random_seed=self.random_seed)
+                )
+
+        return next_batch
+
+    def next_batch(self):
+        """Get the next batch of pipelines to evaluate
+
+        Returns:
+            list(PipelineBase): a list of instances of PipelineBase subclasses, ready to be trained and evaluated.
+        """
+
+        if self._batch_number == 0:
+            next_batch = self._create_naive_pipelines()
+        elif self._batch_number == 1:
+            next_batch = self._create_naive_pipelines_with_feature_selection()
+        elif self._batch_number == 2:
+            next_batch = self._create_fast_final()
+        elif self.batch_number == 3:
+            next_batch = self._create_ensemble()
+        elif self.batch_number == 4:
+            next_batch = self._create_long_top_n(n=3)
+        elif self.batch_number % 2 != 0:
+            next_batch = self._create_ensemble()
+        else:
+            next_batch = []
+            for _ in range(10):
```

Comment:
Im curious if it'd be beneficial to make this for loop its own separate helper function. It could then make the code cleaner/less repetitive for both this spot and when you're finding the 50 instances of each pipeline.